### PR TITLE
Fix golint failures for kubelet/container

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
**What type of PR is this?**

/kind cleanup
/sig node

**What this PR does / why we need it**:

Fix linter issues in kubelet/container component. 

**Which issue(s) this PR fixes**:

Related to #68026.

List of issues:

```
pkg/kubelet/container/container_gc.go:26:1: comment on exported type ContainerGCPolicy should be of the form "ContainerGCPolicy ..." (with optional leading article)
pkg/kubelet/container/container_gc.go:27:6: type name will be used as container.ContainerGCPolicy by other packages, and that stutters; consider calling this GCPolicy
pkg/kubelet/container/container_gc.go:39:1: comment on exported type ContainerGC should be of the form "ContainerGC ..." (with optional leading article)
pkg/kubelet/container/container_gc.go:42:6: type name will be used as container.ContainerGC by other packages, and that stutters; consider calling this GC
pkg/kubelet/container/container_gc.go:67:1: comment on exported function NewContainerGC should be of the form "NewContainerGC ..."
pkg/kubelet/container/helpers.go:105:2: var containerJson should be containerJSON
pkg/kubelet/container/helpers.go:144:1: exported function ExpandContainerVolumeMounts should have comment or be unexported
pkg/kubelet/container/helpers.go:162:1: exported function ExpandContainerCommandAndArgs should have comment or be unexported
pkg/kubelet/container/helpers.go:180:1: comment on exported function FilterEventRecorder should be of the form "FilterEventRecorder ..."
pkg/kubelet/container/helpers.go:223:1: comment on exported function IsHostNetworkPod should be of the form "IsHostNetworkPod ..."
pkg/kubelet/container/helpers.go:228:1: comment on exported function ConvertPodStatusToRunningPod should be of the form "ConvertPodStatusToRunningPod ..."
pkg/kubelet/container/ref.go:29:5: exported var ImplicitContainerPrefix should have comment or be unexported
pkg/kubelet/container/ref.go:29:29: should omit type string from declaration of var ImplicitContainerPrefix; it will be inferred from the right-hand side
pkg/kubelet/container/resize.go:24:1: comment on exported function HandleResizing should be of the form "HandleResizing ..."
pkg/kubelet/container/runtime_cache_fake.go:19:1: comment on exported type TestRuntimeCache should be of the form "TestRuntimeCache ..." (with optional leading article)
pkg/kubelet/container/runtime_cache_fake.go:27:1: exported method TestRuntimeCache.UpdateCacheWithLock should have comment or be unexported
pkg/kubelet/container/runtime_cache_fake.go:33:1: exported method TestRuntimeCache.GetCachedPods should have comment or be unexported
pkg/kubelet/container/runtime_cache_fake.go:39:1: exported function NewTestRuntimeCache should have comment or be unexported
pkg/kubelet/container/runtime_cache.go:29:6: exported type RuntimeCache should have comment or be unexported
pkg/kubelet/container/runtime.go:37:6: exported type Version should have comment or be unexported
pkg/kubelet/container/runtime.go:133:6: exported type ImageService should have comment or be unexported
pkg/kubelet/container/runtime.go:148:6: exported type ContainerAttacher should have comment or be unexported
pkg/kubelet/container/runtime.go:148:6: type name will be used as container.ContainerAttacher by other packages, and that stutters; consider calling this Attacher
pkg/kubelet/container/runtime.go:152:6: exported type ContainerCommandRunner should have comment or be unexported
pkg/kubelet/container/runtime.go:152:6: type name will be used as container.ContainerCommandRunner by other packages, and that stutters; consider calling this CommandRunner
pkg/kubelet/container/runtime.go:185:6: type name will be used as container.ContainerID by other packages, and that stutters; consider calling this ID
pkg/kubelet/container/runtime.go:194:1: exported function BuildContainerID should have comment or be unexported
pkg/kubelet/container/runtime.go:198:1: comment on exported function ParseContainerID should be of the form "ParseContainerID ..."
pkg/kubelet/container/runtime.go:207:1: exported method ContainerID.ParseString should have comment or be unexported
pkg/kubelet/container/runtime.go:221:1: exported method ContainerID.IsEmpty should have comment or be unexported
pkg/kubelet/container/runtime.go:225:1: exported method ContainerID.MarshalJSON should have comment or be unexported
pkg/kubelet/container/runtime.go:229:1: exported method ContainerID.UnmarshalJSON should have comment or be unexported
pkg/kubelet/container/runtime.go:236:1: exported method DockerID.ContainerID should have comment or be unexported
pkg/kubelet/container/runtime.go:243:6: exported type ContainerState should have comment or be unexported
pkg/kubelet/container/runtime.go:243:6: type name will be used as container.ContainerState by other packages, and that stutters; consider calling this State
pkg/kubelet/container/runtime.go:246:2: exported const ContainerStateCreated should have comment (or a comment on this block) or be unexported
pkg/kubelet/container/runtime.go:249:2: comment on exported const ContainerStateUnknown should be of the form "ContainerStateUnknown ..."
pkg/kubelet/container/runtime.go:293:6: type name will be used as container.ContainerStatus by other packages, and that stutters; consider calling this Status
pkg/kubelet/container/runtime.go:335:1: comment on exported method PodStatus.GetRunningContainerStatuses should be of the form "GetRunningContainerStatuses ..."
pkg/kubelet/container/runtime.go:346:1: comment on exported type Image should be of the form "Image ..." (with optional leading article)
pkg/kubelet/container/runtime.go:360:6: exported type EnvVar should have comment or be unexported
pkg/kubelet/container/runtime.go:365:6: exported type Annotation should have comment or be unexported
pkg/kubelet/container/runtime.go:370:6: exported type Mount should have comment or be unexported
pkg/kubelet/container/runtime.go:387:6: exported type PortMapping should have comment or be unexported
pkg/kubelet/container/runtime.go:400:6: exported type DeviceInfo should have comment or be unexported
pkg/kubelet/container/runtime.go:456:6: exported type VolumeMap should have comment or be unexported
pkg/kubelet/container/runtime.go:486:1: receiver name s should be consistent with previous receiver name r for RuntimeStatus
pkg/kubelet/container/runtime.go:511:6: exported type Pods should have comment or be unexported
pkg/kubelet/container/runtime.go:557:1: exported method Pod.FindContainerByID should have comment or be unexported
pkg/kubelet/container/runtime.go:566:1: exported method Pod.FindSandboxByID should have comment or be unexported
pkg/kubelet/container/runtime.go:604:1: comment on exported function BuildPodFullName should be of the form "BuildPodFullName ..."
pkg/kubelet/container/runtime.go:609:1: comment on exported function ParsePodFullName should be of the form "ParsePodFullName ..."
pkg/kubelet/container/runtime.go:622:1: comment on exported type SortContainerStatusesByCreationTime should be of the form "SortContainerStatusesByCreationTime ..." (with optional leading article)
pkg/kubelet/container/sync_result.go:28:1: comment on exported var ErrCrashLoopBackOff should be of the form "ErrCrashLoopBackOff ..."
pkg/kubelet/container/sync_result.go:38:2: exported var ErrRunContainer should have comment or be unexported
pkg/kubelet/container/sync_result.go:48:2: exported var ErrSetupNetwork should have comment or be unexported
pkg/kubelet/container/sync_result.go:57:2: exported const StartContainer should have comment (or a comment on this block) or be unexported
```

**Special notes for your reviewer**:

Issues with "stutters" are left out for a separate PR as they might be more controversial. This is why the component is not removed from the `.golint_failures`.

```
pkg/kubelet/container/container_gc.go:27:6: type name will be used as container.ContainerGCPolicy by other packages, and that stutters; consider calling this GCPolicy
pkg/kubelet/container/container_gc.go:42:6: type name will be used as container.ContainerGC by other packages, and that stutters; consider calling this GC
pkg/kubelet/container/runtime.go:151:6: type name will be used as container.ContainerAttacher by other packages, and that stutters; consider calling this Attacher
pkg/kubelet/container/runtime.go:156:6: type name will be used as container.ContainerCommandRunner by other packages, and that stutters; consider calling this CommandRunner
pkg/kubelet/container/runtime.go:189:6: type name will be used as container.ContainerID by other packages, and that stutters; consider calling this ID
pkg/kubelet/container/runtime.go:254:6: type name will be used as container.ContainerState by other packages, and that stutters; consider calling this State
pkg/kubelet/container/runtime.go:307:6: type name will be used as container.ContainerStatus by other packages, and that stutters; consider calling this Status
```

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by SergeyKanzhelev** [comment]
I signed it

File: pkg/kubelet/container/helpers.go [hunk 1]
```diff
@@ -141,6 +141,7 @@ func ExpandContainerCommandOnlyStatic(containerCommand []string, envs []v1.EnvVa
 	return command
 }
 
+// ExpandContainerVolumeMounts expands the subpath of the given VolumeMount by replacing variable references `with the values of given EnvVar.
```
Comment (mattjmcnaughton) [inline pkg/kubelet/container/helpers.go:144]: Nit: I think there's an unnecessary ` in front of the `with`?

File: pkg/kubelet/container/runtime.go [hunk 2]
```diff
@@ -218,35 +224,43 @@ func (c *ContainerID) String() string {
...
 const (
+	// ContainerStateCreated indicates a container that has been created (e.g. with docker create) but not started.
 	ContainerStateCreated ContainerState = "created"
+	// ContainerStateRunning indicates a currently running container.
 	ContainerStateRunning ContainerState = "running"
-	ContainerStateExited  ContainerState = "exited"
-	// This unknown encompasses all the states that we currently don't care.
+	// ContainerStateExited indicates a container that ran and completed ("stopped" in other contexts, although a created container is technically also "stopped").
+	ContainerStateExited ContainerState = "exited"
+	// ContainerStateUnknown encompasses all the states that we currently don't care (like restarting, paused, dead).
```
Comment (mattjmcnaughton) [inline pkg/kubelet/container/runtime.go:263]: Nit: I think `we currently don't care about` may be more descriptive.

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm
/ok-to-test

Welcome to k8s :)

Thanks for fixing this lint issues.

Some small nits, but otherwise lgtm.

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

Thanks for making the small nit changes :)

**Comment by SergeyKanzhelev** [comment]
No new changes, just rebase

**Pull request merged**